python/tank/deploy/tank_commands/get_entity_commands.py

```diff
@@ -0,0 +1,282 @@
+            "pc_path": {
+                "description": "Path to the pipeline configuration associated "
+                               "with the entities.",
+                "type":        "str"
+            },
+
+            "entities": {
+                "description": """List of entities to fetch the actions for.
+                                  Every entity should be a tuple with the
+                                  following format:
+                                    (entity_type, entity_id)""",
+                "type":        "list"
+            },
+
+            "return_value": {
+                "description": """Dictionary of the commands by entity, with
+                                  the (entity_type, entity_id) tuple used as a
+                                  key. Each value is a list of commands. A
+                                  command is a dictionary with the following
+                                  format:
+                                    {
+                                      "name":  command to execute
+                                      "title": title to display for the command
+                                      "icon":  path to the icon of this command
+                                    }""",
+                "type":        "dict"
+            }
+        }
+
+    _ERROR_CODE_CACHE_OUT_OF_DATE = 1
+    _ERROR_CODE_CACHE_NOT_FOUND = 2
+
+    def run_interactive(self, log, args):
+        """
+        Tank command accessor
+
+        :param log: std python logger
+        :param args: command line args
+        """
+        raise TankError("This Action does not support command line access")
+
+    def run_noninteractive(self, log, parameters):
+        """
+        Tank command API accessor.
+        Called when someone runs a tank command through the core API.
+
+        :param log: std python logger
+        :param parameters: dictionary with tank command parameters
+        """
+        pipeline_config_path = parameters["pc_path"]
+        entities = parameters["entities"]
+
+        # at the moment, the caching mechanism works with the entity types,
+        # so we group by type and fetch the commands for each type
+        per_entity_type = itertools.groupby(entities, operator.itemgetter(0))
+
+        commands_per_entity = {}
+        for (entity_type, entities_of_type) in per_entity_type:
+            # make a list out of the grouped entity tuples
+            entities_of_type = list(entities_of_type)
+            try:
+                cache_content = self._load_cached_data(
+                    pipeline_config_path, entity_type, entities_of_type)
+                commands = self._parse_cached_commands(cache_content)
+
+                # at the moment, the commands are the same for all entities of
+                # that type
+                for entity in entities_of_type:
+                    commands_per_entity[entity] = commands
+            except TankError as e:
```

Comment:
`exception TankError, e:` if you want to be py2.5 compliant and consistent with our code base :)